### PR TITLE
Issue 344 add/delete remote

### PR DESCRIPTION
**Pull request by hsablonniere**
## I'm adding a new command:

- [x] add as a new file in `src/commands`
- [x] add command to `src/index.js`
- [x] add TypeScript definition in `src/index.d.ts`
- [x] update `__tests__/__snapshots__/test-exports.js.snap`
- [x] create a test in `src/__tests__`
- [x] if this is your first time contributing, run `npm run add-contributor` and follow the prompts to add yourself to the README

### CODE REVIEW

**Comment by hsablonniere** [comment]
First WIP proposition based upon the refactoring of #349

**Comment by hsablonniere** [comment]
@wmhilton I rebased my branch over master.

**Comment by hsablonniere** [comment]
## TODO:

- [ ] add command to README list of commands
- [ ] create a new doc page [docs](https://github.com/isomorphic-git/isomorphic-git.github.io/tree/source/docs)/X.md
  - [ ] add page to the [Alphabetical Index](https://github.com/isomorphic-git/isomorphic-git.github.io/blob/source/docs/alphabetic.md)
  - [ ] add page to the [sidebar](https://github.com/isomorphic-git/isomorphic-git.github.io/blob/source/website/sidebars.json)
- [ ] Make decision on the TODOs for `addRemote`:
  - [ ] // TODO do I need to check remote/url args presence?
  - [ ] // TODO we need a specific error, isn't it?
  - [ ] // TODO do we need to set a fetch refspec?
- [ ] Make decision on the TODOs for `deleteRemote`:
  - [ ] // TODO do I need to check remote/url args presence?
  - [ ] // TODO we need a specific error, isn't it?

**Comment by wmhilton** [comment]
Since `isomorphic-git` is mostly used as a library rather than interactively, I decided to differ from git a bit regarding throwing errors.
- `addRemote` will not throw an error if `{remote, url}` matches what's already there. (You still get an error if you try to `addRemote` a `{remote, url}` combination that conflicts with an existing `remote`.)
- `deleteRemote` will not throw an error if `remote` doesn't exist

Basically my rule of thumb is "don't throw an error if the end result is the same as if it had succeeded". Otherwise, people will have to use `listRemotes` and wrap their `deleteRemote` in if statements or add extra try/catches.

`addRemote` also supports a `force: true` option that will let you overwrite the existing remote if there is a conflict.
